pyquil/quil.py

```diff
@@ -75,21 +76,38 @@ def __iter__(self):
 
     def get_qubits(self):
         """
-        :return: a set of all the qubit indices allocated in this program, synthesizing freely
-        allocated qubits if neccessary.
+        Returns all of the qubit indices used in this program, including gate applications and
+        allocated qubits. e.g.
+            >>> p = Program()
+            >>> p.inst(("H", 1))
+            >>> p.get_qubits()
+            {1}
+            >>> q = p.alloc()
+            >>> len(p.get_qubits())
+            2
+
+        :return: A set of all the qubit indices used in this program, synthesizing freely
+         allocated qubits if neccessary.
         :rtype: set
         """
-        allocated_qubits = set()
+        qubits = set()
         self.synthesize()
         for ii, action in self:
             if isinstance(action, Gate):
                 qubit_indices = {qq.index() for qq in action.arguments}
             elif isinstance(action, Measurement):
                 qubit_indices = {action.arguments[0].index()}
+            elif isinstance(action, Instr):
+                qubit_indices = set()
+                for arg in action.arguments:
+                    try:
```

Comment:
Can you change this to isinstance checks instead of a try catch? Makes it easier to read.

Reply:
Sure thing.

Reply:
@stevenheidel I just checked `isinstance(int)`, in particular I'm not checking to see if it's something that I can take the index of if it isn't an `int`. Do you think that's fine?

Reply:
I'd personally do it on both: first, if it's an AbstractQubit (see the issubinstance method in quilbase) then check the index. If it's an int add the int. Otherwise skip.

Btw to check if instance of int you have to use `integer_types` from six so that it works on both 2 and 3.

Reply:
Will do, thanks.